src/main/java/com/alrex/parcool/common/action/impl/Dodge.java

```diff
@@ -127,21 +128,23 @@ public double getSpeedModifier(ActionInfo info) {
 	public boolean canStart(Player player, Parkourability parkourability, IStamina stamina, ByteBuffer startInfo) {
 		boolean enabledDoubleTap = ParCoolConfig.Client.Booleans.EnableDoubleTappingForDodge.get();
 		DodgeDirection direction = null;
+		var dodgeVec = KeyRecorder.getLastMoveVector();
 		if (enabledDoubleTap) {
 			if (KeyRecorder.keyBack.isDoubleTapped()) direction = DodgeDirection.Back;
 			if (KeyRecorder.keyLeft.isDoubleTapped()) direction = DodgeDirection.Left;
 			if (KeyRecorder.keyRight.isDoubleTapped()) direction = DodgeDirection.Right;
 		}
 		if (direction == null && KeyRecorder.keyDodge.isPressed()) {
-            if (KeyBindings.isKeyBackDown()) direction = DodgeDirection.Back;
+			if (KeyBindings.isKeyBackDown()) direction = DodgeDirection.Back;
             if (KeyBindings.isKeyForwardDown()) direction = DodgeDirection.Front;
             if (KeyBindings.isKeyLeftDown()) direction = DodgeDirection.Left;
             if (KeyBindings.isKeyRightDown()) direction = DodgeDirection.Right;
+			if (direction != null) dodgeVec = KeyBindings.getCurrentMoveVector();
 		}
-		if (direction == null) return false;
-        direction = AdditionalMods.betterThirdPerson().handleCustomCameraRotationForDodge(direction);
-        direction = AdditionalMods.shoulderSurfing().handleCustomCameraRotationForDodge(direction);
+		if (direction == null || dodgeVec == null) return false;
 		startInfo.putInt(direction.ordinal());
+		startInfo.putDouble(dodgeVec.x);
+		startInfo.putDouble(dodgeVec.y);
```

Comment:
At `getDodgeVec(Player, ByteBuffer)` this value is restored as X and Z values, is this ok to save X and Y values?